pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -632,25 +631,36 @@ func (m *kubeGenericRuntimeManager) updateContainerResources(pod *v1.Pod, contai
 		pod, containerSpec = restoredPod, restoredContainer
 	}
 
-	var containerResources runtimeapi.LinuxContainerResources
+	memoryLimit := int64(0)
+	cpuShares := int64(0)
 	cpuRequest := int64(0)
 	cpuLimit := int64(0)
-	memoryLimit := int64(0)
+	var containerResources runtimeapi.LinuxContainerResources
+
+	if memoryLimits, found := resources.Limits[v1.ResourceMemory]; found {
+		memoryLimit = memoryLimits.Value()
+	}
+	containerResources.MemoryLimitInBytes = memoryLimit
+
 	if cpuRequests, found := resources.Requests[v1.ResourceCPU]; found {
-		cpuRequest = cpuRequests.Value()
+		cpuRequest = cpuRequests.MilliValue()
 	}
 	if cpuLimits, found := resources.Limits[v1.ResourceCPU]; found {
-		cpuLimit = cpuLimits.Value()
+		cpuLimit = cpuLimits.MilliValue()
 	}
-	if memoryLimits, found := resources.Limits[v1.ResourceMemory]; found {
-		memoryLimit = memoryLimits.Value()
+
+	if cpuRequest == 0 && cpuLimit != 0 {
+		cpuShares = milliCPUToShares(cpuLimit)
+	} else {
+		cpuShares = milliCPUToShares(cpuRequest)
 	}
+	containerResources.CpuShares = cpuShares
 
-	containerResources.CpuShares = int64(cm.MilliCPUToShares(cpuRequest * 1000))
-	cpuQuota, cpuPeriod := cm.MilliCPUToQuota(cpuLimit * 1000)
-	containerResources.CpuPeriod = int64(cpuPeriod)
-	containerResources.CpuQuota = cpuQuota
-	containerResources.MemoryLimitInBytes = memoryLimit
+	if m.cpuCFSQuota {
+		cpuQuota, cpuPeriod := milliCPUToQuota(cpuLimit)
+		containerResources.CpuPeriod = int64(cpuPeriod)
+		containerResources.CpuQuota = cpuQuota
+	}
```

Comment:
please make sure if the if statement is not executed here, UpdateContainerResources will be correct with the default values of CpuPeriod and CpuQuota

Reply:
I verified as mentioned in earlier comment.